### PR TITLE
add command option: folderPrefix

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @herohua @hellosnow @qinezh

### CODE REVIEW

File: src/docfx/Models/BuildCommandOptions.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ internal class BuildCommandOptions : ICanPrintHelpMessage, ILoggable
         [Option("logLevel", HelpText = "Specify to which log level will be logged. By default log level >= Info will be logged. The acceptable value could be Verbose, Info, Warning, Error.")]
         public LogLevel? LogLevel { get; set; }
 
+        [Option("folderPrefix", HelpText = "Specify the folder prefix to docfx config file.")]
```
Comment (hellosnow) [inline src/docfx/Models/BuildCommandOptions.cs]: also update the documentation?

File: src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs [hunk 2]
```diff
@@ -53,9 +54,9 @@ public void WriteLine(ILogItem item)
                 Severity = GetSeverity(level),
                 Message = message,
                 Source = phase,
-                File = file,
+                File = file, // todo : combine with repo root.
                 Line = line,
-                DateTime = DateTime.UtcNow
+                DateTime = DateTime.UtcNow,
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs]: >, [](start = 42, length = 1)

remove?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs]: no keep it, safe to add new property.

File: src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs [hunk 3]
```diff
@@ -10,29 +10,30 @@ namespace Microsoft.DocAsCode.Common
...
             {
                 Directory.CreateDirectory(dir);
             }
             _writer = new StreamWriter(reportPath, true);
+            _repoRoot = repoRoot;
         }
-#endif
 
-        public ReportLogListener(StreamWriter writer)
+        public ReportLogListener(StreamWriter writer, string folderPrefix)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/ReportLogListener.cs]: fixed.

**Pull request merged**